Riot/Modules/Room/Views/BubbleCells/BaseBubbleCell/TimestampDisplayable.swift

```diff
@@ -16,14 +16,8 @@
 
 import UIKit
 
-/// UICollectionViewFlowLayout with right align behavior
-class CollectionViewRightAlignFlowLayout: UICollectionViewFlowLayout {
-
-    override var flipsHorizontallyInOppositeLayoutDirection: Bool {
-        return true
-    }
-
-    override var developmentLayoutDirection: UIUserInterfaceLayoutDirection {
-        return UIUserInterfaceLayoutDirection.rightToLeft
-    }    
+/// `TimestampDisplayable` is a protocol indicating that a view support displaying a timestamp.
+@objc protocol TimestampDisplayable {
```

Comment:
I would've named it `TimestampLabelDisplayProtocol` 😁 

Perhaps now is the right time to let cells build their own accessory labels based on their own style. I would envision this protocol having a pair of `displayTimestamp(_ timestamp: Date)`/`hideTimestamp()` methods for displaying arbitrary timestamps with self-defined UI.

The same applies to the `BubbleCellThreadSummaryDisplayable` in which cells conforming to it can build their own `ThreadSummaryView` instead of receiving it from the outside (or a completely different one based on the case).

Reply:
Yes I understand your POV, in fact in this case we can use the suffix `-able` because we add the capacity to the cell to display timestamp. If it was a protocol to abstract the whole implementation of the cell like `RoomTimelineCell` in this case it's ok to use `RoomTimelineCellProtocol`.

Yes I see what you mean regarding decorations. At the moment we are using the same decorations for every cells and cell style. Plus we don't want to make too much data logic in the cell. But yes having a cleaner "viewData" for cells can be a great start. For sure we still have improvement to make to these cells :)